apps/prairielearn/src/ee/lib/ai-grading/ai-grading.ts

```diff
@@ -145,9 +146,23 @@ export async function aiGrade({
     });
     job.info(`Found ${instance_questions.length} submissions to grade!`);
 
+    const gradingJobMapping = await selectGradingJobsInfo(instance_questions);
+
     let error_count = 0;
     // Grade each instance question
     for (const instance_question of instance_questions) {
+      const grading_jobs = gradingJobMapping[instance_question.id] ?? [];
+
+      const manualGradingJob = grading_jobs.find((job) => job.grading_method === 'Manual');
+      const aiGradingJob = grading_jobs.find((job) => job.grading_method === 'AI');
+
+      const should_update_score =
+        !manualGradingJob || // no manual grading job
+        (manualGradingJob &&
+          aiGradingJob &&
+          manualGradingJob.graded_at &&
+          aiGradingJob.graded_at &&
```

Comment:
Nit, you can simplify a bit:

```suggestion
        (manualGradingJob?.graded_at &&
          aiGradingJob?.graded_at &&
```

Reply:
Not needed anymore after the logic change in https://github.com/PrairieLearn/PrairieLearn/pull/12319/commits/c7aa10af3c8992a78e97a860776dbce941572f12